client/client.go

```diff
@@ -159,11 +169,14 @@ func (c *GRPCClient) WithTraceID(ctx context.Context, id string) context.Context
 	return metadata.NewOutgoingContext(ctx, md)
 }
 
-func authContext(ctx context.Context) context.Context {
+func (c *GRPCClient) withAuthToken(ctx context.Context) context.Context {
+	if c.authToken != "" {
+		return metadata.NewOutgoingContext(ctx, metadata.Pairs(apiTokenKey, c.authToken))
+	}
+
 	token := os.Getenv(apiTokenEnvVarName)
```

Comment:
here the behavior is that env var always takes priority even if the token is empty string. Wouldn't that then not reset the context api token?
Scenario:
Env var is set to token of one endpoint, client need to talk to another Dapr sidecar that does not have any authentication.

Reply:
> here the behavior is that env var always takes priority even if the token is empty string

It isn't. Look at it closely, `if c.authToken != ""` then the `metadata.Pairs` method is using the value of `c.authToken` and returns, else continues in the method...

Explicit token set by user always take priority to enable the user to create multiple instances of the client pointing to any number/combination of Dapr instances whether they are authenticating public APIs or not.

Reply:
thanks for reaching out @markrussinovich and pointing out the need to remove (override) the env var set value. Logic adjusted. PTAL